Updating SC docker docs (#1212)
saucelabs/sauce-docs: docs/secure-connections/sauce-connect/setup-configuration/ci-cd-integration.md

* If you'd prefer to manage Docker image tags instead of Sauce Connect Proxy versions.
* If your setup involves several instances running on the same system, Docker would simplify Sauce Connect Proxy port management.

### Running the SC Docker Image

1. Before running the container, you'll need to pull the Sauce Connect Proxy Docker Image from the Docker Hub. This will always pull the latest version of Sauce Connect Proxy.
  ```bash
  $ docker pull saucelabs/sauce-connect
  ```
   * Or - if you _do_ want to use specific SC version - you can specify that as a tag:
   ```bash
   $ docker pull saucelabs/sauce-connect:4.7.1
   ```
2. To run the Sauce Connect Proxy Docker image, execute the below script, which will also set your Sauce Labs username and access key as [environment variables](/basics/environment-variables/).
  ```bash
  $ export SAUCE_USERNAME="my-user"
  $ export SAUCE_ACCESS_KEY="my-access-key"
  docker run \
      -e SAUCE_USERNAME=${SAUCE_USERNAME} \
      -e SAUCE_ACCESS_KEY=${SAUCE_ACCESS_KEY} \
      -it saucelabs/sauce-connect
  ```

If desired, you can specify any additional [SC CLI arguments](/dev/cli/sauce-connect-proxy/) here.

> If your tests are using localhost addresses, you should also set `--network="host"` as an argument in the above script. Otherwise Sauce Connect within the Docker container will not be able to access your local services in the host machine.

Reviewer: Specify this is for Linux hosts only

* If you'd prefer to manage Docker image tags instead of Sauce Connect Proxy versions.
* If your setup involves several instances running on the same system, Docker would simplify Sauce Connect Proxy port management.

### Running the SC Docker Image

1. Before running the container, you'll need to pull the Sauce Connect Proxy Docker Image from the Docker Hub. This will always pull the latest version of Sauce Connect Proxy.
  ```bash
  $ docker pull saucelabs/sauce-connect
  ```
   * Or - if you _do_ want to use specific SC version - you can specify that as a tag:
   ```bash
   $ docker pull saucelabs/sauce-connect:4.7.1
   ```
2. To run the Sauce Connect Proxy Docker image, execute the below script, which will also set your Sauce Labs username and access key as [environment variables](/basics/environment-variables/).
  ```bash
  $ export SAUCE_USERNAME="my-user"
  $ export SAUCE_ACCESS_KEY="my-access-key"
  docker run \
      -e SAUCE_USERNAME=${SAUCE_USERNAME} \
      -e SAUCE_ACCESS_KEY=${SAUCE_ACCESS_KEY} \
      -it saucelabs/sauce-connect
  ```

If desired, you can specify any additional [SC CLI arguments](/dev/cli/sauce-connect-proxy/) here.

> ::::note
> In Linux, if your tests use localhost addresses, `--network="host"` is required as an argument in the above script to allow requests originating within the container to access the Docker host machine.
> ::::

Reviewer 2: ```suggestion
:::note
If you're running tests in Linux that use localhost addresses, you'll need to specify `--network="host"` as an argument in the above script to allow requests originating within the container to access the Docker host machine.
:::
```

> ::::warning
> `--network="host"` behaves differently on Windows, Mac OS, and Linux due to the respective platform Docker implementation.
> ::::

Reviewer 2: We reserve **Warning** admonition boxes for major alerts that could cause data loss or a breaking change

```suggestion
:::info
`--network="host"` behaves differently on Windows, macOS, and Linux due to the respective platform Docker implementation.
:::
```

### Running the Sauce Connect Proxy Docker Image with a CI/CD Pipeline
If you want to run this Docker image as part of your CI/CD pipeline, you would need a way to determine that Sauce Connect Proxy is ready to proxy the requests. You can achieve that by:

> 1. Starting with v4.8.0, the `/readiness` endpoint is available.
> 1. By using Docker's [volumes](https://docs.docker.com/storage/volumes/) feature and [--readyfile](/dev/cli/sauce-connect-proxy/#--readyfile).

Reviewer 2: ```suggestion
- Using the `/readiness` endpoint, which is available effective with Sauce Connect Proxy v4.8.0
- Using Docker's [volumes](https://docs.docker.com/storage/volumes/) feature along with our Sauce Connect Proxy [`--readyfile`](/dev/cli/sauce-connect-proxy/#--readyfile) flag.
```

#### Readiness Endpoint

> ::::note
> Only available starting with Sauce Connect Proxy version 4.8.0.
> ::::
> 
> Starting with 4.8.0, Sauce Connect Proxy Docker image allows configuring liveness and readiness HTTP probes. See the [kubernetes documentation](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/) for more info.

Reviewer 2: ```suggestion
The readiness endpoint, supported in Sauce Connect Proxy versions 4.8.0 and higher, allows you to configure liveness and readiness HTTP probes. See the [Kubernetes documentation](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/) for more information.
```

> Docker container exposes Sauce Connect Proxy HTTP status server on port 8032. The following endpoints are available:

Reviewer 2: ```suggestion
When you use one of the following endpoints, the Docker container will expose the Sauce Connect Proxy HTTP status server on port 8032:
```

Author: no, docker ALWAYS exposes that port, not just "When you use one of the following endpoints"

> * /readiness returns 200 response code when Sauce Connect Proxy is ready, 503 otherwise

Reviewer 2: ```suggestion
* `/readiness` returns 200 response code when Sauce Connect Proxy is ready, 503 otherwise
```

> * /liveness returns 200 response code when Sauce Connect Proxy is running

Reviewer 2: ```suggestion
* `/liveness` returns 200 response code when Sauce Connect Proxy is running
```

> You can leverage the "readiness" endpoint in our CI/CD pipeline by running the following:

Reviewer 2: ```suggestion
You can leverage the readiness endpoint in our CI/CD pipeline by running the following:
```

1. Create a simple bash script `wait-for-sc.sh` that will ensure the pipeline only continues after Sauce Connect Proxy is fully connected and ready.
   ```bash title="wait-for-sc.sh"
   until [ "$(curl -s -o /dev/null -w ''%{http_code}'' localhost:8032/readiness)" == "200" ]
   do
       sleep 2
   done
   echo "SC ready"
   ```
1. Run Sauce Connect Docker container using the script below. It is important that you map port 8032 so that the port is available to the host.
  ```bash
  $ docker run \
      -e SAUCE_USERNAME=${SAUCE_USERNAME} \
      -e SAUCE_ACCESS_KEY=${SAUCE_ACCESS_KEY} \
      -p 8032:8032 \
      -t saucelabs/sauce-connect:latest \
      -i some-identifier --detach
    $ ./wait-for-sc.sh
    ```

::::note
In this example, using `--network="host"` renders the port mapping unnecessary.
::::

#### Ready File

> You can leverage the Sauce Connect Proxy [--readyfile](/dev/cli/sauce-connect-proxy/#--readyfile) flag that allows to specify a file that will be created (or updated) when the proxy is ready.

Reviewer 2: ```suggestion
You can leverage the Sauce Connect Proxy [`--readyfile`](/dev/cli/sauce-connect-proxy/#--readyfile) flag to specify a file that will be created (or updated) when the proxy is ready.
```

1. Create a simple bash script `wait-for-sc.sh` that will ensure the pipeline only continues after Sauce Connect Proxy is fully connected and ready.
   ```bash title="wait-for-sc.sh"
   until [ -f /tmp/sc.ready ]
   do
       sleep 1
   done
   echo "SC ready"
   ```
1. Run Sauce Connect Docker container using the script below. It is important that you mount a temp folder here so that `wait-for-sc.sh` can detect when Sauce Connect has launched.
  ```bash
  $ docker run \
      -e SAUCE_USERNAME=${SAUCE_USERNAME} \
      -e SAUCE_ACCESS_KEY=${SAUCE_ACCESS_KEY} \
      -v /tmp:/tmp \
      --network="host" \
      -t saucelabs/sauce-connect:latest \
      -f /tmp/sc.ready \
      -i some-identifier --detach
    $ ./wait-for-sc.sh
  ```
::::note
`--network="host"` allows Sauce Connect Proxy to access your app in the host machine. It's only required if your app runs on the same machine as the docker host.
::::

> Starting with Sauce Connect Proxy 4.8.0, the "ready" file will contain a JSON-formatted information about the running Sauce Connect Proxy. For example:

Reviewer 2: ```suggestion
Starting with Sauce Connect Proxy 4.8.0, the "ready" file will contain aJSON-formatted information about the running Sauce Connect Proxy. For example:
```
